### ISSUE TITLE
[Bug]: Video Vimeo and Dailymotion not work

### ISSUE THREAD

**Issue by mediafant** [pimcore/pimcore#10152]
### Expected behavior

At Video-Output from Object Youtube works, but vimeo and dailymotion-ids produce an error

On the demopage at
 https://demo.pimcore.com/admin/login/deeplink?object_1121_object
i insert vimeolink, but the whole eventpages didn't work at:
https://demo.pimcore.com/en/Events

Also the documentation:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Video_Type.html

```
with this lines are not correct:
$video->type = $v->getType();
    $video->id = ($videoData instanceof Asset) ? $videoData->getId() : $videoData;
    $video->title = $v->getTitle();
    $video->description = $v->getDescription();
    if($v->getPoster()) {
        $video->poster = $v->getPoster()->getId();
}

```
Solution for this is like this:

```
//$videoData
$data['type'] = $v->getType();
$data['id'] = ($videoData instanceof Asset) ? $videoData->getId() : $videoData;
$data['title'] = $v->getTitle();
$data['description'] = $v->getDescription();
if($v->getPoster()) {
  $data['poster'] = $v->getPoster()->getId();
} else {
  $data['poster'] = '';
}
$video->setDataFromEditmode($data);
```

For working vimeo and dailymotion it is necessary to change code in:
/pimcore/models/Document/Editable/Video.php
in methodes getDailymotionCode() and getVimeoCode() you have to add a little but code:

```
isset($config['config']['clip']) && 
```
to this:
```
if (isset($config['config']['clip']) && is_array($config['config']['clip'])) {
                $clipConfig = $config['config']['clip'];
}
            $configurations = $clipConfig;
            if (array_key_exists('vimeo', $config) && is_array($config['vimeo'])) {
                $configurations = array_merge($clipConfig, $config['vimeo']);
            }
```
like in getYoutubeCode()

But in getHtml5Code() there is a problem too. There is no Error-Message but didn't work.
With one mpg-Video there is a problem on this line:
```
$duration = ceil($video->getDuration());
```
... because there is no duration in $video. With other Videofile there is no message, but video didn't start.

### Actual behavior

No error-message but only preview-image is visible, no working video

### Steps to reproduce

Perhaps on this page:
https://demo.pimcore.com/en/Events

on "Auto Fair Salzburg '19" I added an vimeo-video, but all links on this page didn't work

**Comment by brusch** [comment]
@mediafant Thanks for reporting the issue - since we're quite busy and you have already debugged the issue, would it be possible for you to provide us the fixes as a PR? Thanks a lot!

**Comment by dvesh3** [comment]
@mediafant the page you are referring to specifically uses Youtube example **https://github.com/pimcore/demo/blob/beb4a68c76351f17a6f048b30f71899da66be0a3/templates/default/events.html.twig#L94.** could you please share a generic reproducible scenario? thanks!

**Comment by dvesh3** [comment]
🏓 @mediafant

**Comment by mediafant** [comment]
@dvesh3 - don't understand - youtube works. But vimeo-links and dailymotion-links didn't work. So I changed pimcore-code so it works.
 But there is also a problem on pimcore-demo-page at:
 https://demo.pimcore.com/en/Events
 Buttons "More Details" didn't work

**Comment by dvesh3** [comment]
I mean is it reproducible with `{{ pimcore_video('sample_video'}}` on frontend template?

> But there is also a problem on pimcore-demo-page at:
> https://demo.pimcore.com/en/Events
> Buttons "More Details" didn't work

this is a separate issue and fixed with commit https://github.com/pimcore/demo/commit/1c0c1d0f0fe2fc5234f3ece442f9c14746c8d6e9

**Comment by dvesh3** [comment]
Fixed by #10370
